private_gpt/server/chat/chat_service.py

```diff
@@ -97,6 +100,7 @@ def _chat_engine(
         system_prompt: str | None = None,
         use_context: bool = False,
         context_filter: ContextFilter | None = None,
+        context_template: str | None = DEFAULT_CONTEXT_TEMPLATE,
```

Comment:
The default value should be injected in the body of the function instead of here (given that it is coming from `settings()`) to make testing easier.

Reply:
Ok, something like:
```
def _chat_engine(
        self,
        system_prompt: str | None = None,
        use_context: bool = False,
        context_filter: ContextFilter | None = None,
        context_template: str | None = DEFAULT_CONTEXT_TEMPLATE,
    ) -> BaseChatEngine:
        if use_context:
            if context_template is None:
                context_template = DEFAULT_CONTEXT_TEMPLATE  #(or alternatively just settings().rag.default_context_template)
```

Reply:
It should be 

```
if context_template is None: context_template = settings.rag.default_context_template
```

where `settings` is injected in the constructor and remove `DEFAULT_CONTEXT_TEMPLATE`